spec.html

```diff
@@ -314,22 +315,29 @@ <h1>SharedArrayBuffer.prototype.grow ( _newLength_ )</h1>
         1. Let _newByteLength_ to ? ToIntegerOrInfinity(_newLength_).
         1. Let _rawCurrentByteLengthBytesRead_ be a List of length 8 whose elements are nondeterministically chosen byte values.
         1. NOTE: In implementations, _rawCurrentByteLengthBytesRead_ is the result of a load-link, of a load-exclusive, or of an operand of a read-modify-write instruction on the underlying hardware. The nondeterminism is a semantic prescription of the memory model to describe observable behaviour of hardware with weak consistency.
+        1. Let _byteLengthBlock_ be _O_.[[ArrayBufferByteLengthData]].
         1. Let _isLittleEndian_ be the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
         1. Let _currentBytesLength_ be RawBytesToNumeric(~BigUint64~, _rawCurrentByteLengthBytesRead_, _isLittleEndian_).
-        1. If _newByteLength_ &lt; _currentByteLength_ or _newByteLength_ &gt; _O_.[[ArrayBufferMaxByteLength]], throw a *RangeError* exception.
-        1. NOTE: The above may fail due to concurrent calls to SharedArrayBuffer.prototype.grow racing. For example, if one agent is evaluating `sab.grow(10)` and another agent is evaluating `sab.grow(20)` concurrently, either may fail due to a race while atomically updating the length.
-        1. If _newByteLength_ &ne; _currentByteLength_, then
+        1. Let _growFailed_ be *false*.
+        1. If _newByteLength_ &lt; _currentByteLength_ or _newByteLength_ &gt; _O_.[[ArrayBufferMaxByteLength]], set _growFailed_ to *true*.
+        1. Let _byteLengthDelta_ be _newByteLength_ - _currentByteLength_.
+        1. If it is impossible to create a new Shared Data Block value consisting of _byteLengthDelta_ bytes, set _growFailed_ to *true*.
+        1. NOTE: While the observable behavior of growable SharedArrayBuffers is specified by allocating a max-sized Shared Data Block at construction time no new Shared Data Block is constructed, this step is intended to capture the requirement that implementations that run out of memory must throw a *RangeError*.
```

Comment:
> max-sized Shared Data Block at construction time no new Shared Data Block is constructed

maybe two versions of this sentence got edited together?

Reply:
oh wow, how did this happen